docarray/index/backends/weaviate.py

```diff
@@ -111,6 +111,17 @@ def __init__(self, db_config=None, **kwargs) -> None:
         self._set_properties()
         self._create_schema()
 
+    @property
+    def index_name(self):
+        default_index_name = self._schema.__name__ if self._schema is not None else None
+        if default_index_name is None:
+            raise ValueError(
+                'A WeaviateDocumentIndex must be typed with a Document type.'
+                'To do so, use the syntax: WeaviateDocumentIndex[DocumentType]'
+            )
```

Comment:
I don't think that this code is reachable because basically the same check happens already earlier, but maybe I am wrong

Reply:
yes I know, but `mypy` complains otherwise